keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -35,5 +40,30 @@ class ArtifactListener(
     }
   }
 
+  @EventListener(ArtifactRegisteredEvent::class)
+  fun onArtifactRegisteredEvent(event: ArtifactRegisteredEvent) {
+    val artifact = event.artifact
+    if (artifactRepository.isRegistered(artifact.name, artifact.type)) {
+      log.debug("Artifact {} is already registered", artifact)
+      return
+    }
+    artifactRepository.register(artifact)
+    runBlocking {
+      artifactService
+        .getVersions(artifact.name)
+        .firstOrNull()
```

Comment:
That's definitely the right order, right? It's not the kind of mistake I'd ever make, but I thought it was worth double-checking.

Reply:
😆 Yep, from this endpoint the versions come back sorted with newest first.